source/agora/utils/Log.d

```diff
@@ -198,6 +206,49 @@ public class CircularAppender : Appender
     }
 }
 
+///
+unittest
+{
+    class MockLayout : Appender.Layout
+    {
+        /// Format the message
+        public override void format (LogEvent event, scope FormatterSink dg)
+        {
+            sformat(dg, "{}", event);
+        }
+    }
+
+    auto get_log_output = (const string log_msg){
```

Comment:
```suggestion
    scope get_log_output = (string log_msg){
```

`const string` implicitly converts to `string`, and the other way around, so no need to clog the prototype with a constraints that will only apply to the function's body.